### DISCUSSION TITLE
[FE] type 선언이 많아짐에 따른 타입 선언 분리

### DISCUSSION THREAD

**Discussion by 2yunseong** [woowacourse-teams/2023-stamp-crush#628] [General]
타입 선언이 많아짐에 따라 타입선언을 분리해야하는 필요성이 생겼습니다. 어떻게 분리하면 좋을까요?

**Comment by regularPark** [comment]
현재와 같이 통신/그 외 두 가지로 나누어지는 큰 방향성은 동일하나,
하위에 `Cafe`, `Coupon`, `Customer`와 같이 같은 카테고리라 생각되는 것들을 묶어주는 것은 어떨까요?

**Reply by 2yunseong** [reply]
좋습니다. 필요해보입니다!

**Comment by kangyeongmin** [comment]
types/domain.ts - 우리 domain에 설계된 table을 기반으로 이름 및 속성들을 정의해줍니다. (Cafe, Customer, Owner, Coupon 등...)

types/api/response.ts - suffix가 Res인 타입들을 모아 response타입들의 관심사를 모아줍니다.
types/api/request.ts - Req와 관련된 타입들을 모아줍니다. 상단에 MutateReq -> Parmas -> body 순으로 선언합니다.

**Reply by 2yunseong** [reply]
라잇 방식에 실제로 앱에서 사용하는 타입들은 API에서 추출해서 사용하는건 어떨까요?
base가 되는 Type에서, Omit, Pick 등을 이용해서 재선언 해주는 방식은 어떠신가요??
